src/platform/with_wgl/native_gl_context.rs

```diff
@@ -0,0 +1,144 @@
+}
+
+impl Drop for NativeGLContext {
+    fn drop(&mut self) {
+        unsafe {
+            if !self.weak {
+                wgl::DeleteContext(self.render_ctx as *const _);
+                let window = user32::WindowFromDC(self.device_ctx);
+                user32::ReleaseDC(window, self.device_ctx);
+                user32::DestroyWindow(window);
+            }
+        }
+    }
+}
+
+pub struct NativeGLContextHandle(pub winapi::HGLRC, pub winapi::HDC);
+unsafe impl Send for NativeGLContextHandle {}
+
+impl NativeGLContextMethods for NativeGLContext {
+    type Handle = NativeGLContextHandle;
+
+    fn get_proc_address(addr: &str) -> *const () {
+        let addr = CString::new(addr.as_bytes()).unwrap().as_ptr();
+        unsafe {
+            let p = wgl::GetProcAddress(addr) as *const _;
+            if !p.is_null() { return p; }
+            match *GL_LIB {
+                Some(ref lib) =>  kernel32::GetProcAddress(lib.0, addr) as *const _,
+                None => ptr::null_mut()
+            }
+        }
+
+    }
+
+    fn create_shared(with: Option<&Self::Handle>) -> Result<NativeGLContext, &'static str> {
+        match unsafe{super::utils::create_offscreen(with, &WGLAttributes::default())}{
+            Ok(ctx) => {
+                //wglGetProcAddress only works in the presence of a valid GL context
+                //OpenGL functions must be loaded after the first context is created
+                ctx.make_current().unwrap();
+                load_gl();
+                Ok(ctx)
+            }
+            Err(s) => {
+                error!("WGL: {}", s);
+                Err("Error creating WGL context")
+            }
+        }
+    }
+
+    fn is_current(&self) -> bool {
+        unsafe { wgl::GetCurrentContext() == self.render_ctx as *const c_void }
+    }
+
+    fn current() -> Option<Self> {
+        if let Some(handle) = Self::current_handle() {
+            Some(NativeGLContext {
+                render_ctx: handle.0,
+                device_ctx: handle.1,
+                weak: true
+            })
+        }
+        else {
+            None
+        }
+    }
+
+    fn current_handle() -> Option<Self::Handle> {
+        let handle = unsafe{ wgl::GetCurrentContext()};
+        if !handle.is_null() {
+            let hdc = unsafe { wgl::GetCurrentDC()};
+            Some(NativeGLContextHandle(handle as winapi::HGLRC, hdc as winapi::HDC))
+        }
+        else {
+            None
+        }
+    }
+
+    fn make_current(&self) -> Result<(), &'static str> {
+         if unsafe {wgl::MakeCurrent(self.device_ctx as * const _, self.render_ctx as *const _) != 0 } {
+            Ok(())
+        } else {
+            Err("wgl::makeCurrent failed")
+        }
+    }
+
+    fn unbind(&self) -> Result<(), &'static str> {
+        if self.is_current() {
+            unsafe {wgl::MakeCurrent(ptr::null_mut(),ptr::null_mut() );}
+            Ok(())
+        }
+        else {
+            Err("gwl::MakeCurrent (on unbind)")
```

Comment:
We definitely shouldn't return an error if the current context was already unbound, the error condition should be `wgl::MakeCurrent` failing.